src/d3a/models/area/event_types.py

```diff
@@ -0,0 +1,43 @@
+
+
+class IntervalAreaEvent:
+    def __init__(self, disconnect_start, disconnect_end):
+        self.disconnect_start = disconnect_start
+        self.disconnect_end = disconnect_end
+        self._active = True
+
+    @property
+    def active(self):
+        return self._active
+
+    def tick(self, current_time):
+        self._active = not self.disconnect_start <= current_time.hour < self.disconnect_end
+
+
+class DisableIntervalAreaEvent(IntervalAreaEvent):
+    pass
+
+
+class DisconnectIntervalAreaEvent(IntervalAreaEvent):
```

Comment:
Aren't DisconnectIntervalAreaEvent & DisableIntervalAreaEvent redundant and be achieved via IntervalAreaEvent

Reply:
Not really, because these are different events, disableinterval is for blackouts, disconnectinterval is for islanding.